mapper/tedge_mapper/src/az_mapper.rs

```diff
@@ -0,0 +1,198 @@
+use crate::error::*;
+use crate::mapper::*;
+use crate::time_provider::TimeProvider;
+use mqtt_client::Topic;
+use thin_edge_json::serialize::ThinEdgeJsonSerializer;
+
+#[derive(Debug)]
+pub struct AzureMapperConfig {
+    pub in_topic: Topic,
+    pub out_topic: Topic,
+    pub errors_topic: Topic,
+}
+
+impl Default for AzureMapperConfig {
+    fn default() -> Self {
+        Self {
+            in_topic: Topic::new("tedge/measurements").unwrap(),
+            out_topic: Topic::new("az/messages/events/").unwrap(),
+            errors_topic: Topic::new("tedge/errors").unwrap(),
+        }
+    }
+}
+
+impl Into<MapperConfig> for AzureMapperConfig {
+    fn into(self) -> MapperConfig {
+        MapperConfig {
+            in_topic: self.in_topic,
+            out_topic: self.out_topic,
+            errors_topic: self.errors_topic,
+        }
+    }
+}
+
+pub struct AzureConverter {
+    pub(crate) add_timestamp: bool,
+    pub(crate) time_provider: Box<dyn TimeProvider>,
+}
+
+impl Converter for AzureConverter {
+    type Error = ConversionError;
+    fn convert(&self, input: &[u8]) -> Result<Vec<u8>, Self::Error> {
+        // Size check
+        let threshold = 255 * 1000; // 255KB
```

Comment:
you could make the `threshold` a field of `AzureConverter` (`size_threshold`). That would actually allow you to more easily test it, by setting it to a very low number and trying to convert a small JSON. At 255 kb... you need to generate a rather huge JSON... and that complicates tests a bit (well, okay, not much.... you can easily create a large JSON by just create a very large string).

Reply:
Just curious: Is the threshold really`255 KB` or is it `255 KiB`?

Reply:
Should be `256KB` 
```suggestion
        let threshold = 256 * 1024; // 256KB
```

Reply:
Make sense to make the `threshold` a field of `AzureConverter`. 
Testing 255KB JSON, which I avoided in unit tests... should we have it? 

The threshold is 255KB. Azure's max message size is [256KB](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-d2c-guidance), so I make it a bit smaller than the actual threshold as Azure will add some fields to a payload.